scripts/globals/abilities/pets/somnolence.lua

```diff
@@ -0,0 +1,41 @@
+---------------------------------------------------
+-- Somnolence
+---------------------------------------------------
+
+require("scripts/globals/settings");
+require("scripts/globals/status");
+require("scripts/globals/monstertpmoves");
+require("scripts/globals/magic");
+
+---------------------------------------------------
+
+function onAbilityCheck(player, target, ability)
+    return 0,0;
+end;
+
+function onPetAbility(target, pet, skill)
+    local spell = getSpell(165); --gave the ability the same power as Thunder II
+    local dmg = calculateMagicDamage(178,1,pet,spell,target,ELEMENTAL_MAGIC_SKILL,MOD_INT,false);
+    local dInt = pet:getStat(MOD_INT) - target:getStat(MOD_INT);
+    local resist = applyPlayerResistance(pet,-1,target, dInt ,ELEMENTAL_MAGIC_SKILL, ELE_DARK);
```

Comment:
Your code is all squished making it hard to read but it looks like you variablized resist then  tossed that into another variable named gravResist and thats not needed because the original resist variable is never modified?? Gonna talk to you on IRC.